install.py

```diff
@@ -42,47 +42,79 @@ def download(url, to):
     subprocess.run(curl_cmd, check=True)
 
 
-def main():
+def check_platform():
     if sys.platform == 'darwin':
         if os.geteuid() == 0:
             print("This script is intended to run as non-root on macOS")
-            sys.exit(1)
+            return 1
         if not available("brew"):
             print("Please install Homebrew before continuing install on macOS")
-            sys.exit(2)
+            return 2
     elif sys.platform == 'linux':
         if os.geteuid() != 0:
             print("This script is intended to run as root on Linux")
-            sys.exit(3)
+            return 3
     else:
         print("This script is intended to run on Linux and macOS only")
-        sys.exit(4)
+        return 4
+
+    return 0
+
+
+def install_mac_dependencies():
+    subprocess.run(["pip3", "install", "--break-system-packages",
+                   "huggingface_hub[cli]==0.24.2"], check=True)
+    subprocess.run(
+        ["pip3", "install", "--break-system-packages", "omlmd==0.1.4"], check=True)
+    subprocess.run(["brew", "install", "llama.cpp"], check=True)
+
+
+def setup_ramalama(bindir, tmp_dir):
+    binfile = "ramalama"
+    from_file = binfile + ".py"
+    host = "https://raw.githubusercontent.com"
+    branch = os.getenv('BRANCH', 's')
+    url = f"{host}/containers/ramalama/{branch}/{from_file}"
+    to_file = os.path.join(tmp_dir, from_file)
+    download(url, to_file)
+    if sys.platform == 'darwin':  # macOS
```

Comment:
How about:
```
ramalama_bin = os.path.join(bindir, binfile)
syspath = '/usr/share/ramalama'
if sys.platform == 'darwin':  # macOS
        install_mac_dependencies()
        sharedirs = ["/opt/homebrew/share",
                     "/usr/local/share"]
        syspath = next((d for d in sharedirs if os.path.exists(d)), None)
        syspath += "/ramalama"
```

Reply:
Yeah sure, one less comparison would be nice, will catch in the morning and merge